cmd/kubeadm/app/preflight/preflight.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"k8s.io/kubernetes/pkg/util/procfs"
+)
+
+// TODO(phase1+): Check for data already in /var/lib/etcd, /etc/kubernetes or /var/lib/kubelet and warn that it can fail because of stale data there
+// We also would want to check if kubelet is running, but it's hard because it's running some milliseconds only when crashlooping
+// We would have to add systemctl-specific logic and that might not be good.
+func Verify() error {
+	if err := requireRoot(); err != nil {
+		return err
+	}
+	if binariesNotFound, err := requireBinaries("kubelet", "docker"); err != nil {
+		return fmt.Errorf("<preflight> The required binaries %v were not found in PATH. Please install them and retry.", binariesNotFound)
+	}
+	if binariesNotFound, err := requireBinaries("iptables", "ip", "nsenter", "mount", "umount", "socat"); err != nil {
+		fmt.Printf("<preflight warning> The for kubelet required binaries %v were not found in PATH. You may have to install them in order for Kubernetes to function properly.\n", binariesNotFound)
+	}
+	if dockererr, dockerderr := requireProcess("docker"), requireProcess("dockerd"); dockererr != nil && dockerderr != nil {
+		return fmt.Errorf("<preflight> docker or dockerd have to be running on this machine in order for kubeadm to start. Please start docker and try again.")
+	}
+	return nil
```

Comment:
Yes, containerised kubelet is one of the use-cases we want to support. Additionally, user might not have kubelet in the path, or anything like this may be possible.